lerobot/common/policies/vla/modeling_vla.py

```diff
@@ -0,0 +1,308 @@
+    """
+
+    name = "vla"
+
+    def __init__(
+        self,
+        config: VLAConfig,
+        dataset_stats: dict[str, dict[str, Tensor]] | None = None,
+    ):
+        """
+        Initialize the VLAPolicy class with model configuration.
+
+        Args:
+        config (VLAConfig): Configuration for the Qwen2VL model.
+        """
+        super().__init__()
+        if config is None:
+            config = VLAConfig()
+        self.config: VLAConfig = config
+
+        self.normalize_inputs = Normalize(
+            config.input_shapes, config.input_normalization_modes, dataset_stats
+        )
+        self.normalize_targets = Normalize(
+            config.output_shapes, config.output_normalization_modes, dataset_stats
+        )
+        self.unnormalize_outputs = Unnormalize(
+            config.output_shapes, config.output_normalization_modes, dataset_stats
+        )
+        
+        self.language_model = LlavaOnevisionForConditionalGeneration.from_pretrained("llava-hf/llava-onevision-qwen2-7b-ov-hf", torch_dtype=torch.float16, device_map = 'cuda')
+        self.device= self.language_model.device
+        self.model = VLA(config).to(self.device)
+        self.processor = AutoProcessor.from_pretrained("llava-hf/llava-onevision-qwen2-7b-ov-hf")# Updated Qwen2VL without loss and lm_head
+        
+
+        self.expected_image_keys = [k for k in config.input_shapes if k.startswith("observation.image")]
+
+        self.reset()
+
+    def reset(self):
+        """This should be called whenever the environment is reset."""
+        self._action_queue = deque([], maxlen=self.config.n_action_steps)
+
+    @torch.no_grad()
+    def select_action(self, batch: dict[str, torch.Tensor]) -> torch.Tensor:
+        self.eval()
+        #batch = self.normalize_inputs(batch)
+        
+        if len(self.expected_image_keys) > 0:
+            batch = dict(batch)  
+            batch["observation.images"] = torch.stack([batch[k] for k in self.expected_image_keys], dim=-4).to(self.device)
+        
+        batch["prompt"] = self.config.prompt.to(self.device)
+        
+        # Process inputs (text and images) using the processor
+        processed_inputs = self.processor(
+            text=batch["prompt"], videos=list(batch["observation.images"]), return_tensors="pt", padding=True, do_rescale=False
+        ).to(self.device)
+
+        processed_inputs["pixel_values_videos"] = processed_inputs["pixel_values_videos"].to(self.device).to(torch.float16)
+        breakpoint()
+        # Forward pass through Llava (to get hidden states)
+        llava_output = self.language_model(  # Calling the Llava model inside VLA
+            **processed_inputs,
+            return_dict=True,
+            output_hidden_states=True
+        )
+        
+        hidden_states = llava_output.hidden_states[-1]  # Use last layer's hidden state
+        hidden_states = hidden_states[:, -4:, :] #make 4 a config parameter 
```

Comment:
Taking the last 4 embeddings as an input to the action decoder. Because chunk_size is divisible by 4.